### ISSUE TITLE
(incremental, check-unsat-cores) Segmentation fault on QF_LRA formula 

### ISSUE THREAD

**Issue by muchang** [cvc5/cvc5#4925]
Hi,
For this formula, CVC4  throws out a segmentation fault:
```
[624] % cvc4 small.smt2
sat
unsat
CVC4 suffered a segfault.
Offending address is 0x30
Looks like a NULL pointer was dereferenced.
Aborted
[625] % 
[625] % cat small.smt2
(set-logic QF_LRA)
(set-option :incremental true)
(set-option :check-unsat-cores true)
(declare-fun a () Real)
(declare-fun b () Real)
(declare-fun c () Real)
(assert (< c 0))
(assert (> (+ a b) 0))
(assert (or (< a 1) (> c 1)))
(check-sat)
(assert (= b (- 1)))
(check-sat true)
(check-sat)
[626] %
```

OS: Ubuntu 18.04
Commit: c110363

**Comment by nafur** [comment]
With debug: `cvc4: /home/gereon/cvc4_master/src/prop/minisat/core/Solver.cc:1389: void CVC4::Minisat::Solver::removeClausesAboveLevel(CVC4::Minisat::vec<unsigned int>&, int): Assertion !locked(c) failed.` on
```
(set-option :incremental true)
(set-option :check-unsat-cores true)
(declare-fun a () Real)
(declare-fun b () Real)
(declare-fun c () Real)
(assert (= c 0))
(assert (= 0 (+ a b)))
(assert (or (< a 1) (= c 1)))
(check-sat)
(assert (= b (- 1)))
(check-sat true)
```

Looks like a duplicate of #3455.

**Comment by HanielB** [comment]
This is indeed a duplicate of #3455. As an internal note, this failure also happens in proof-new due to the known issue of clauses being added at the wrong level in incremental. When that is fixed and we start using the new proof infrastructure for generating unsat cores we can close this issue.

**Comment by zhendongsu** [comment]
Another related instance; looks like the test from https://github.com/cvc5/cvc5/issues/4925#issuecomment-717131794 doesn't reproduce anymore: 

```
[733] % cvc4 -q -i --check-unsat-cores small.smt2
sat
unsat
Fatal failure within void cvc5::Minisat::Solver::removeClausesAboveLevel(cvc5::Minisat::vec<unsigned int>&, int) at /local/suz-local/software/CVC4/src/prop/minisat/core/Solver.cc:1590
Check failure

 !locked(c)

Aborted
[734] % 
[734] % cat small.smt2
(declare-fun a () Real)
(declare-fun b () Real)
(declare-fun c () Real)
(assert (or (= c 0) (= a 0)))
(assert (= c b 0))
(check-sat)
(push)
(assert (not (= c 0)))
(check-sat)
(pop)
(check-sat)
[735] % 
```

**Comment by HanielB** [comment]
Closed via #6574
